src/pgduckdb_node.cpp

```diff
@@ -99,28 +102,45 @@ ExecuteQuery(DuckdbScanState *state) {
 	auto &prepared = *state->prepared_statement;
 	auto &query_results = state->query_results;
 	auto &connection = state->duckdb_connection;
-	auto pg_params = state->params;
-	const auto num_params = pg_params ? pg_params->numParams : 0;
+	auto paramLI = state->params;
+	List *paramlist = NIL;
 	duckdb::vector<duckdb::Value> duckdb_params;
-	for (int i = 0; i < num_params; i++) {
-		ParamExternData *pg_param;
-		ParamExternData tmp_workspace;
 
-		/* give hook a chance in case parameter is dynamic */
-		if (pg_params->paramFetch != NULL) {
-			pg_param = pg_params->paramFetch(pg_params, i + 1, false, &tmp_workspace);
-		} else {
-			pg_param = &pg_params->params[i];
-		}
-
-		if (pg_param->isnull) {
-			duckdb_params.push_back(duckdb::Value());
-		} else if (OidIsValid(pg_param->ptype)) {
-			duckdb_params.push_back(pgduckdb::ConvertPostgresParameterToDuckValue(pg_param->value, pg_param->ptype));
-		} else {
-			std::ostringstream oss;
-			oss << "parameter '" << i << "' has an invalid type (" << pg_param->ptype << ") during query execution";
-			throw duckdb::Exception(duckdb::ExceptionType::EXECUTOR, oss.str().c_str());
+	/*
+	 * Let's recursively iterate through the relevant parameters now,
+	 * and then deal with the parameter values later
+	 */
+	(void) Get_ParamList((Node *)(state->query), (void **) (&paramlist));
+	if (list_length(paramlist)) {
+#if PG_VERSION_NUM >= 150000
+		foreach_node(Param, param, paramlist) {
+#else
+	 	foreach_ptr(Param, param, paramlist) {
+#endif
+			if (param->paramkind == PARAM_EXTERN &&
+				paramLI != NULL && param->paramid > 0 &&
+				param->paramid <= paramLI->numParams) {
+				ParamExternData *prm;
+				ParamExternData prmdata;
```

Comment:
Can you revert the naming changes for these two? It makes it quite hard to see what actual code changes you made if you also change the naming of existing variables at the same time.